__tests__/registries/npm-registry.js

```diff
@@ -39,3 +39,122 @@ describe('normalizeConfig', () => {
     expect(normalized).toEqual(rooted);
   });
 });
+
+describe('request', () => {
+  function createMocks(): Object {
+    const mockRequestManager = {
+      request: jest.fn(),
+    };
+    const mockRegistries = {
+      npm: jest.fn(),
+      yarn: {
+        getScopedOption: jest.fn(),
+      },
+    };
+    const mockReporter = jest.fn();
+
+    return {
+      mockRequestManager,
+      mockRegistries,
+      mockReporter,
+    };
+  }
+
+  test('should call requestManager.request with pathname url', () => {
+    const testCwd = '.';
+    const {mockRequestManager, mockRegistries, mockReporter} = createMocks();
+    const npmRegistry = new NpmRegistry(testCwd, mockRegistries, mockRequestManager, mockReporter);
+
+    const pathname = 'http://github.com/yarnpkg/yarn.tgz';
+
+    npmRegistry.request(pathname);
+
+    const requestParams = mockRequestManager.request.mock.calls[0][0];
+
+    expect(requestParams.url).toBe(pathname);
```

Comment:
You should probably use [toHaveBeenLastCalledWith](https://facebook.github.io/jest/docs/expect.html#tohavebeenlastcalledwitharg1-arg2-) instead.